sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -247,9 +247,37 @@ export class MessageSender extends LinkEntity {
    * @param sendBatch Boolean indicating whether the encoded message represents a batch of messages or not
    * @return {Promise<Delivery>} Promise<Delivery>
    */
-  private _trySend(encodedMessage: Buffer, sendBatch?: boolean): Promise<void> {
+  private _trySend(
+    encodedMessage: Buffer,
+    sendBatch?: boolean,
+    options?: GetSenderOptions
+  ): Promise<void> {
+    if (!options) {
+      options = {};
+    }
     const sendEventPromise = () =>
       new Promise<void>(async (resolve, reject) => {
+        if (!this.isOpen()) {
```

Comment:
Shouldn't the timer for `actionAfterTimeout` start before this `if` check so that the time taken for initializing is included in the per try timeout?

Also, the `actionAfterTimeout` is set to run after `Constants.defaultOperationTimeoutInMs`. Shouldn't we be using the timeout from the retry options instead?

Reply:
Oh yes, you're right. Updated both.